packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -124,8 +124,9 @@ async function createAppRouteCode({
 
     resolvedPagePath = `next-metadata-route-loader?${stringify({
       page,
+      filePath: resolvedPagePath,
       isDynamic: isDynamic ? '1' : '0',
-    })}!${resolvedPagePath}${`?${WEBPACK_RESOURCE_QUERIES.metadataRoute}`}`
+    })}!?${WEBPACK_RESOURCE_QUERIES.metadataRoute}`
```

Comment:
Kinda confused by the `!` here... is that still needed as we don't even need a resource path?

Reply:
We need a `!` to identify this is a loader request otherwise it cannot be resolved